signingscript/src/signingscript/rcodesign.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+"""Functions that interface with rcodesign"""
+import asyncio
+import logging
+from signingscript.exceptions import SigningScriptError
+
+log = logging.getLogger(__name__)
+
+
+class RCodesignError(SigningScriptError):
+    pass
+
+
+async def _execute_command(command):
+    """Executes a command, logging output, and return exitcode and output lines
+    Args:
+        command (str): The command to be run
+
+    Returns:
+        (Tuple) exit code, log lines
+    """
+    log.info("Running command: {}".format(" ".join(command)))
+    proc = await asyncio.create_subprocess_exec(
+        *command,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+    log.info("COMMAND OUTPUT: ")
+    output_lines = []
+    while True:
+        # If at EOF, stop
+        if proc.stdout.at_eof() and proc.stderr.at_eof():
+            break
+        # Handle stdout
+        stdout = (await proc.stdout.readline()).decode("utf-8").rstrip()
+        if stdout:
+            log.info(stdout)
+            output_lines.append(stdout)
+        # Handle stderr
+        stderr = (await proc.stderr.readline()).decode("utf-8").rstrip()
+        if stderr:
+            # Unfortunately a lot of outputs from rcodesign come out to stderr
+            log.warn(stderr)
+            output_lines.append(stderr)
```

Comment:
I agree with @jcristau's earlier comments about mixing stdout and stderr (this is just another form of that). Presumably you know whether it's stdout or stderr that has the string you're looking for, so you should be able to keep these separate?

Reply:
@jcristau comment was regarding sending `stderr=asyncio.subprocess.stdout` to `asyncio.create_subprocess_exec` - which wouldn't allow us to collect them at all.

But, as stated on the comment above the `log.warn`, `rcodesign`doesn't only output errors to stderr (there's good data in it). Therefore separating them has the opposite desired effect.
Let me know if the inline comment isn't enough.